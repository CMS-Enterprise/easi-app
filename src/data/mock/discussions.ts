import {
  SystemIntakeGRBReviewDiscussionFragment,
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

const mockDiscussions = (
  systemIntakeID: string = '0d191c27-4de3-46d8-9ded-c2e9c6b3fbd6'
): SystemIntakeGRBReviewDiscussionFragment[] => [
  {
    __typename: 'SystemIntakeGRBReviewDiscussion',
    initialPost: {
      __typename: 'SystemIntakeGRBReviewDiscussionPost',
      id: '882357e4-c0b0-44ef-b749-f71879ad7878',
      content:
        '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>',
      votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
      grbRole: SystemIntakeGRBReviewerRole.SUBJECT_MATTER_EXPERT,
      createdByUserAccount: {
        __typename: 'UserAccount',
        id: '034fa2b3-00ff-4ec6-857e-75291a59df74',
        commonName: 'Adeline Aarons'
      },
      systemIntakeID,
      createdAt: '2024-11-12T21:59:07.368862Z'
    },
    replies: [
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '4099dbb7-2752-4bf9-a3e1-da225ceb9fae',
        content:
          '<p>Nisi nobis consectetur voluptatem neque tempore. Ea nihil sed beatae?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.NON_VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CMCS_REP,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '909a7888-4d6f-4bbf-9b9d-71ec1e2c3068',
          commonName: 'Jerry Seinfeld'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      },
      {
        __typename: 'SystemIntakeGRBReviewDiscussionPost',
        id: '47b0081d-de33-4514-b68f-a7e2bbf2610f',
        content:
          '<p>Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?</p>',
        votingRole: SystemIntakeGRBReviewerVotingRole.VOTING,
        grbRole: SystemIntakeGRBReviewerRole.CO_CHAIR_CIO,
        createdByUserAccount: {
          __typename: 'UserAccount',
          id: '601d52be-7baa-4b45-91cd-88b4a5935c3f',
          commonName: 'Cosmo Kramer'
        },
        systemIntakeID,
        createdAt: '2024-11-13T21:59:07.368862Z'
      }
    ]
  }
];

export default mockDiscussions;
