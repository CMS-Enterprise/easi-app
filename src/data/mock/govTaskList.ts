import {
  ITGovDecisionStatus,
  ITGovDraftBusinessCaseStatus,
  ITGovFeedbackStatus,
  ITGovFinalBusinessCaseStatus,
  ITGovGRBStatus,
  ITGovGRTStatus,
  ITGovIntakeFormStatus
} from 'types/graphql-global-types';
import { GetGovernanceTaskListWithMockData } from 'types/itGov';

const id = 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa';

/** IT Gov Task List status states */
// eslint-disable-next-line import/prefer-default-export
export const taskListState: {
  [k: string]: GetGovernanceTaskListWithMockData;
} = {
  /** Event: Intake Request form: Requester starts a new request */
  intakeFormNotStarted: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.READY,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.CANT_START,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      submittedAt: null,
      updatedAt: null
    }
  },
  /** Event: Intake Request form: Requester starts Intake Request form */
  intakeFormInProgress: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.IN_PROGRESS,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.CANT_START,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      submittedAt: null,
      updatedAt: null
    }
  },
  /** Event: Intake Request form: Requester submits Intake Request form */
  intakeFormSubmitted: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.IN_REVIEW,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      submittedAt: '2023-07-07T00:30:28Z',
      updatedAt: null
    }
  },
  /** Event: Intake Request form: Admin takes the action to request edits to the Intake Request form */
  intakeFormEditsRequested: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.EDITS_REQUESTED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.COMPLETED,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [
        {
          __typename: 'GovernanceRequestFeedback',
          id
        }
      ],
      submittedAt: '2023-07-07T00:30:28Z',
      updatedAt: '2023-07-08T00:30:28Z'
    }
  },
  /** Event: Intake Request form: Requester re- submits Intake Request form */
  intakeFormResubmittedAfterEdits: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.IN_REVIEW,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [
        {
          __typename: 'GovernanceRequestFeedback',
          id
        }
      ],
      submittedAt: '2023-07-09T00:30:28Z',
      updatedAt: '2023-07-09T00:30:28Z'
    }
  },

  feedbackFromInitialReviewCantStart: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.IN_PROGRESS,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.CANT_START,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      submittedAt: null,
      updatedAt: null
    }
  },
  feedbackFromInitialReviewInProgress: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.IN_REVIEW,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      submittedAt: null,
      updatedAt: null
    }
  },
  feedbackFromInitialReviewDoneNoFeedback: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.COMPLETED,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [],
      governanceRequestFeedbackCompletedAt: '2023-07-10T00:30:28Z',
      submittedAt: null,
      updatedAt: null
    }
  },
  feedbackFromInitialReviewDoneWithFeedback: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.COMPLETED,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [
        {
          __typename: 'GovernanceRequestFeedback',
          id
        }
      ],
      governanceRequestFeedbackCompletedAt: '2023-07-10T00:30:28Z',
      submittedAt: null,
      updatedAt: null
    }
  },
  feedbackFromInitialReviewResubmittedWithFeedback: {
    systemIntake: {
      __typename: 'SystemIntake',
      id,
      itGovTaskStatuses: {
        __typename: 'ITGovTaskStatuses',
        intakeFormStatus: ITGovIntakeFormStatus.COMPLETED,
        feedbackFromInitialReviewStatus: ITGovFeedbackStatus.IN_REVIEW,
        decisionAndNextStepsStatus: ITGovDecisionStatus.CANT_START,
        bizCaseDraftStatus: ITGovDraftBusinessCaseStatus.CANT_START,
        grtMeetingStatus: ITGovGRTStatus.CANT_START,
        bizCaseFinalStatus: ITGovFinalBusinessCaseStatus.CANT_START,
        grbMeetingStatus: ITGovGRBStatus.CANT_START
      },
      governanceRequestFeedbacks: [
        {
          __typename: 'GovernanceRequestFeedback',
          id
        }
      ],
      submittedAt: null,
      updatedAt: null
    }
  }
};
