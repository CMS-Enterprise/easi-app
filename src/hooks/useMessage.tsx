import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
