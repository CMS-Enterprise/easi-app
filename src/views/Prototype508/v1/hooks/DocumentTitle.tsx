import { useEffect, useRef } from 'react';

function useDocumentTitle(title: string) {
  const defaultTitle = useRef(document.title);

  useEffect(() => {
    document.title = title;
  }, [title]);

  useEffect(() => {
    const newTitle = defaultTitle.current;
    return () => {
      if (newTitle) {
        document.title = newTitle;
      }
    };
  }, []);
}

export default useDocumentTitle;
