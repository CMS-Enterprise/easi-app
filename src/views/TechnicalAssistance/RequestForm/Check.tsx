import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';

import Pager from './Pager';
import { FormStepComponentProps } from '.';

function Check({ request, stepUrl, taskListUrl }: FormStepComponentProps) {
  const { t } = useTranslation('technicalAssistance');
  const history = useHistory();
  return (
    <Pager
      back={{
        onClick: () => {
          history.push(stepUrl.back);
        }
      }}
      next={{
        onClick: () => {
          history.push(stepUrl.next);
        },
        text: t('check.submit')
      }}
      taskListUrl={taskListUrl}
    />
  );
}

export default Check;
