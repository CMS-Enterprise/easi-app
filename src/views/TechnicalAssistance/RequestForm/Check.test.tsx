import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { ApolloQueryResult, NetworkStatus } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';

import { requester } from 'data/mock/trbRequest';
import { MessageProvider } from 'hooks/useMessage';
import {
  GetTrbRequest,
  GetTrbRequest_trbRequest as TrbRequest,
  GetTrbRequestVariables
} from 'queries/types/GetTrbRequest';
import {
  TRBAdviceLetterStatus,
  TRBAttendConsultStatus,
  TRBCollabGroupOption,
  TRBConsultPrepStatus,
  TRBFeedbackStatus,
  TRBFormStatus,
  TRBRequestState,
  TRBRequestType,
  TRBSubjectAreaOption,
  TRBWhereInProcessOption
} from 'types/graphql-global-types';

import Check from './Check';

const mockTrbRequestData: TrbRequest = {
  id: 'f3b4cff8-321d-4d2a-a9a2-4b05810756d7',
  name: 'Lorem ipsum dolor sit amet, consectetur',
  state: TRBRequestState.OPEN,
  taskStatuses: {
    __typename: 'TRBTaskStatuses',
    formStatus: TRBFormStatus.IN_PROGRESS,
    feedbackStatus: TRBFeedbackStatus.READY_TO_START,
    consultPrepStatus: TRBConsultPrepStatus.READY_TO_START,
    attendConsultStatus: TRBAttendConsultStatus.READY_TO_SCHEDULE,
    adviceLetterStatus: TRBAdviceLetterStatus.READY_TO_START
  },
  feedback: [],
  type: TRBRequestType.NEED_HELP,
  form: {
    id: '452cf444-69b2-41a9-b8ab-ed354d209307',
    component: 'CCSQ',
    needsAssistanceWith:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed maecenas gravida tristique maecenas. Id id ipsum, purus ac ornare. A, ut et et, sollicitudin turpis sit porta. Enim, dictumst eu vulputate et lacus et habitant. Sit quisque gravida condimentum augue erat mauris metus, arcu. Malesuada posuere fames integer sed eu tortor vel. Non scelerisque elementum auctor urna consectetur. Ut eget hendrerit massa pharetra pellentesque dolor risus in. In pellentesque vitae ac porttitor amet lacinia id. Cursus amet tortor posuere pharetra et augue eros, lorem vitae. Sit viverra cursus cras consequat, ut amet.',
    hasSolutionInMind: true,
    proposedSolution:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed maecenas gravida tristique maecenas. Id id ipsum, purus ac ornare. A, ut et et, sollicitudin turpis sit porta. Enim, dictumst eu vulputate et lacus et habitant. Sit quisque gravida condimentum augue erat mauris metus, arcu. Malesuada posuere fames integer sed eu tortor vel. Non scelerisque elementum auctor urna consectetur. Ut eget hendrerit massa pharetra pellentesque dolor risus in.',
    whereInProcess: TRBWhereInProcessOption.OTHER,
    whereInProcessOther: 'A different brainstorm',
    hasExpectedStartEndDates: true,
    expectedStartDate: '2023-01-05T05:00:00Z',
    expectedEndDate: null,
    systemIntakes: [],
    collabGroups: [TRBCollabGroupOption.SECURITY, TRBCollabGroupOption.OTHER],
    collabDateSecurity: '06/01/2023',
    collabDateEnterpriseArchitecture: null,
    collabDateCloud: null,
    collabDatePrivacyAdvisor: null,
    collabDateGovernanceReviewBoard: null,
    collabDateOther: 'Lorem ipsum dolor',
    collabGroupOther: 'Consectetur',
    collabGRBConsultRequested: null,
    subjectAreaOptions: [
      TRBSubjectAreaOption.ACCESSIBILITY_COMPLIANCE,
      TRBSubjectAreaOption.CLOUD_MIGRATION
    ],
    subjectAreaOptionOther: 'A few other subjects',
    fundingSources: null,
    submittedAt: '2023-01-23T20:06:52.123703Z',
    __typename: 'TRBRequestForm'
  },
  __typename: 'TRBRequest'
};

const mockRefetch = async (
  variables?: Partial<GetTrbRequestVariables> | undefined
): Promise<ApolloQueryResult<GetTrbRequest>> => {
  return {
    loading: false,
    networkStatus: NetworkStatus.ready,
    data: {
      trbRequest: mockTrbRequestData
    }
  };
};

describe('Trb Request form: Check and submit', () => {
  it('renders request form field values', async () => {
    const mockStore = configureMockStore();
    const store = mockStore({
      auth: {
        euaId: requester?.userInfo?.euaUserId,
        name: requester?.userInfo?.commonName
      }
    });
    const { asFragment, getByText, getByRole } = render(
      <MemoryRouter>
        <MockedProvider>
          <Provider store={store}>
            <MessageProvider>
              <Check
                request={mockTrbRequestData}
                stepUrl={{
                  current:
                    '/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/basic',
                  next:
                    '/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/subject',
                  back:
                    '/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/undefined'
                }}
                refetchRequest={mockRefetch}
                setIsStepSubmitting={() => {}}
                setStepSubmit={() => {}}
                setFormAlert={() => {}}
                taskListUrl=""
              />
            </MessageProvider>
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    // Check some strings for the correct formatting
    // Submitted date
    getByText('January 23, 2023');
    // Request type
    getByText('Iâ€™m having a problem with my system');
    // Where in process "other" field
    getByText('Other: A different brainstorm');
    // Expected start date
    getByText('Yes, 01/05/2023 expected start');
    // Oit groups with other
    getByText('Security (06/01/2023), Other: Consectetur (Lorem ipsum dolor)');
    // Net sec options
    getByText('Access Control and Identity Management');
    // App dev with other
    getByText('Cloud Migration');
    // Others with no topics
    getByText('A few other subjects');

    // Snapshot of stuff in place
    expect(asFragment()).toMatchSnapshot();

    // Submit Request is available
    const submitButton = getByRole('button', { name: 'Submit request' });
    expect(submitButton).not.toBeDisabled();
  });
});
