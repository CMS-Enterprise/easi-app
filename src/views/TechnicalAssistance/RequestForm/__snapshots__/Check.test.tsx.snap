// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trb Request form: Check and submit renders request form field values 1`] = `
<DocumentFragment>
  <dl
    class="easi-dl margin-top-4 margin-bottom-3"
  >
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Submission date
        </dt>
        <dd>
          January 23, 2023
        </dd>
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Request type
        </dt>
        <dd>
          Iâ€™m having a problem with my system
        </dd>
      </div>
    </div>
  </dl>
  <div
    class="easi-divider"
  />
  <h2
    class="margin-top-3 margin-bottom-05"
  >
    Basic request details
  </h2>
  <div>
    <a
      class="usa-link"
      href="/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/basic"
    >
      Edit this section
    </a>
  </div>
  <dl
    class="easi-dl margin-y-3"
  >
    <div
      class="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Request name
        </dt>
        <dd>
          Lorem ipsum dolor sit amet, consectetur
        </dd>
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Request component
        </dt>
        <dd>
          CCSQ
        </dd>
      </div>
      <div
        class="grid-col-12"
        data-testid="grid"
      >
        <dt>
          What do you need technical assistance with?
        </dt>
        <dd>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed maecenas gravida tristique maecenas. Id id ipsum, purus ac ornare. A, ut et et, sollicitudin turpis sit porta. Enim, dictumst eu vulputate et lacus et habitant. Sit quisque gravida condimentum augue erat mauris metus, arcu. Malesuada posuere fames integer sed eu tortor vel. Non scelerisque elementum auctor urna consectetur. Ut eget hendrerit massa pharetra pellentesque dolor risus in. In pellentesque vitae ac porttitor amet lacinia id. Cursus amet tortor posuere pharetra et augue eros, lorem vitae. Sit viverra cursus cras consequat, ut amet.
        </dd>
      </div>
      <div
        class="grid-col-12"
        data-testid="grid"
      >
        <dt>
          Do you have a solution in mind already?
        </dt>
        <dd>
          Yes
        </dd>
        <dt>
          Describe your proposed solution.
        </dt>
        <dd>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed maecenas gravida tristique maecenas. Id id ipsum, purus ac ornare. A, ut et et, sollicitudin turpis sit porta. Enim, dictumst eu vulputate et lacus et habitant. Sit quisque gravida condimentum augue erat mauris metus, arcu. Malesuada posuere fames integer sed eu tortor vel. Non scelerisque elementum auctor urna consectetur. Ut eget hendrerit massa pharetra pellentesque dolor risus in.
        </dd>
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Where are you in your process?
        </dt>
        <dd>
          Other: A different brainstorm
        </dd>
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Does your solution have an expected start and/or end date?
        </dt>
        <dd>
          Yes, 01/05/2023 expected start  
        </dd>
      </div>
      <div
        class="tablet:grid-col-12"
        data-testid="grid"
      >
        <dt>
          Which existing funding sources will fund this project?
        </dt>
      </div>
      <div
        class="grid-row tablet:grid-col-12 margin-bottom-4"
        data-testid="grid"
      >
        No answer selected
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          List any Life Cycle IDs (LCIDs) pertaining to this request.
        </dt>
        <dd>
          No answer selected
        </dd>
      </div>
      <div
        class="desktop:grid-col-12"
        data-testid="grid"
      />
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      >
        <dt>
          Select any other OIT groups that you have met with or collaborated with.
        </dt>
        <dd>
          Security (06/01/2023), Other: Consectetur (Lorem ipsum dolor)
        </dd>
      </div>
      <div
        class="tablet:grid-col-12 desktop:grid-col-6"
        data-testid="grid"
      />
    </div>
  </dl>
  <div
    class="easi-divider"
  />
  <h2
    class="margin-top-3 margin-bottom-05"
  >
    Subject areas
  </h2>
  <div>
    <a
      class="usa-link"
      href="/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/subject"
    >
      Edit this section
    </a>
  </div>
  <div
    class="margin-top-3"
  >
    <span
      class="easi-tag text-base-darker bg-base-lighter margin-bottom-1"
      data-testid="tag"
    >
      Access Control and Identity Management
    </span>
    <span
      class="easi-tag text-base-darker bg-base-lighter margin-bottom-1"
      data-testid="tag"
    >
      Cloud Migration
    </span>
  </div>
  <dl
    class="easi-dl margin-y-3"
  >
    <dt>
      Other subject areas
    </dt>
    <dd>
      A few other subjects
    </dd>
  </dl>
  <div
    class="easi-divider"
  />
  <h2
    class="margin-top-3 margin-bottom-05"
  >
    Attendees
  </h2>
  <div>
    <a
      class="usa-link"
      href="/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/attendees"
    >
      Edit this section
    </a>
  </div>
  <div
    class="margin-top-3 margin-bottom-6"
  >
    <span
      class="font-body-2xs"
    >
      There are no attendees
    </span>
  </div>
  <div
    class="easi-divider"
  />
  <h2
    class="margin-top-3 margin-bottom-05"
  >
    Supporting documents
  </h2>
  <div>
    <a
      class="usa-link"
      href="/trb/requests/f3b4cff8-321d-4d2a-a9a2-4b05810756d7/documents"
    >
      Edit this section
    </a>
  </div>
  <div
    class="margin-top-3 margin-bottom-6"
  >
    <div
      class="text-center"
    >
      <span
        aria-busy="true"
        aria-valuetext="Loading"
        class="easi-spinner easi-spinner--xl"
        role="progressbar"
      />
    </div>
  </div>
  <div
    class="grid-row grid-gap"
    data-testid="grid"
  >
    <div
      class="tablet:grid-col-12 desktop:grid-col-6 margin-top-2 margin-bottom-4"
      data-testid="grid"
    >
      <div
        class="usa-summary-box"
        data-testid="summary-box"
      >
        <div
          class="usa-summary-box__body"
        >
          <h3
            class="usa-summary-box__heading"
          >
            What happens next?
          </h3>
          <div
            class="usa-summary-box__text"
          >
            <div>
              The TRB will review and get back to you with one of these outcomes:
            </div>
            <ul
              class="usa-list margin-y-05"
            >
              <li>
                schedule a consult session
              </li>
              <li>
                direct you to another governance process or team
              </li>
              <li>
                request additional information
              </li>
            </ul>
            <div>
              A TRB team member will get back to you within two business days.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="border-base-light border-top-1px"
  >
    <ul
      class="usa-button-group margin-top-2"
    >
      <li
        class="usa-button-group__item"
      >
        <div
          class="display-flex flex-align-center"
        >
          <button
            class="usa-button usa-button--outline margin-top-0 margin-bottom-1 mobile-lg:margin-bottom-0"
            data-testid="button"
            type="button"
          >
            Back
          </button>
        </div>
      </li>
      <li
        class="usa-button-group__item"
      >
        <div
          class="display-flex flex-align-center"
        >
          <button
            class="usa-button margin-top-0"
            data-testid="button"
            type="submit"
          >
            Submit request
          </button>
        </div>
      </li>
    </ul>
    <button
      class="usa-button usa-button--unstyled margin-top-2 display-flex flex-align-center"
      data-testid="button"
      type="button"
    >
      <svg
        class="usa-icon margin-right-05"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      Save and exit
    </button>
  </div>
</DocumentFragment>
`;
